### PR TITLE
Fix commas in hexadecimal

### PR DESCRIPTION
**Pull request by dan-simon**

### CODE REVIEW

**Comment by Razenpok** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/antimatter-dimensions/notations/pullRequest?prid=6047753&bid=19813270)

File: src/utils.ts [hunk 1]
```diff
@@ -2,12 +2,20 @@ import Decimal from "break_infinity.js";
 
 const commaRegexp = /\B(?=(\d{3})+(?!\d))/gu;
 
+const higherBaseCommaRegexp = /\B(?=([0-9A-Za-z]{3})+(?![0-9A-Za-z]))/gu;
```
Comment (Razenpok) [inline src/utils.ts:5]: Codacy found an issue: [Unsafe Regular Expression](https://app.codacy.com/gh/antimatter-dimensions/notations/pullRequest?prid=6047753)

**Comment by Razenpok** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/antimatter-dimensions/notations/pullRequest?prid=6047753&bid=19813270)

File: src/utils.ts [hunk 2]
```diff
@@ -1,10 +1,27 @@
...
+  if (index === 0) {
+    return value.slice(-3);
+  } else {
+    return value.slice(-3 * (index + 1), -3 * index);
+  }
+}
+
+function addCommas(value: string): string {
+  return Array.from(Array(Math.ceil(value.length / 3))).map(
+    (_, i) => commaSection(value, i)).reverse().join(',');
```
Comment (Razenpok) [inline src/utils.ts:13]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/antimatter-dimensions/notations/pullRequest?prid=6047753)

**Comment by razenpok** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/antimatter-dimensions/notations/pullRequest?prid=6047753&bid=19813270)

**Pull request merged**